pkg/errors/norequeue_error.go

```diff
@@ -48,3 +50,51 @@ func IsNoRequeueNoError(err error) bool {
 func NoRequeueNoErr(msg string) error {
 	return NoRequeueError{Message: msg, DoNotErr: true}
 }
+
+// AggregateOrNoRequeue aggregates the given errors and returns a NoRequeueError
+// if all the errors are NoRequeueError and set doNotErr to false
+// if one of them has DoNotErr: false.
+func AggregateOrNoRequeue(errs []error) error {
+	if len(errs) == 0 {
+		return nil
+	}
+
+	var (
+		flattenErrs  []error
+		allNoRequeue = true
+		doNotErr     = true
+	)
+
+	for _, err := range errs {
+		// Aggregate error does not support Errors.As()
+		if agg, ok := err.(apierrorsutil.Aggregate); ok { //nolint:errorlint
+			flattenErrs = append(flattenErrs,
+				apierrorsutil.Flatten(agg).Errors()...)
+		} else {
+			flattenErrs = append(flattenErrs, err)
```

Comment:
I think you'd also want check for the `Unwrap() []error` interface here too?

I think it might be clearer to instead of flatten, walk & check as we go so instead of building up this flatten list to find like the first one isn't NoReq.

Reply:
removed the helper function.